src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -592,20 +616,56 @@ protected SnapshotResult<YugabyteDBOffsetContext> doExecute(ChangeEventSourceCon
    * @param tabletsWaitingForCallback a set containing tablets which have completed snapshot from server but have not received the explicit checkpoint
    */
   public void doSnapshotCompletionCheck(YBPartition partition, Set<String> snapshotCompletedTablets,
-                                        Set<String> tabletsWaitingForCallback) {
-      OpId opId = OpId.from(this.tabletToExplicitCheckpoint.get(partition.getId()));
-      if (opId == null) {
+                                        Set<String> tabletsWaitingForCallback, YugabyteDBOffsetContext offsetContext) throws Exception {
+      if (this.tabletToExplicitCheckpoint.get(partition.getId()) == null) {
         // If we have no OpId stored in the explicit checkpoint map then that would indicate that
         // we haven't yet received any callback from Kafka even once and we should wait more.
         return;
       }
 
-      if (isSnapshotCompleteMarker(opId)) {
+      OpId opId = OpId.from(this.tabletToExplicitCheckpoint.get(partition.getId()));
+      if (isSnapshotCompleteMarker(opId) && !snapshotCompletedTablets.contains(partition.getId())) {
```

Comment:
is the 2nd condition required? It is also checked before calling this fn. Maybe an assert is better instead of checking twice? 

Specifically this condition `snapshotCompletedTablets.contains`